python/src/main.cpp

```diff
@@ -24,31 +26,35 @@ opAlgoDispatchPyInit(std::shared_ptr<kp::Algorithm>& algorithm,
                  push_consts.size(),
                  std::string(py::str(push_consts.dtype())));
 
-    if (push_consts.dtype().is(py::dtype::of<std::float_t>())) {
-        std::vector<float> dataVec((float*)info.ptr,
-                                   ((float*)info.ptr) + info.size);
-        return std::unique_ptr<kp::OpAlgoDispatch>{ new kp::OpAlgoDispatch(
-          algorithm, dataVec) };
-    } else if (push_consts.dtype().is(py::dtype::of<std::uint32_t>())) {
-        std::vector<uint32_t> dataVec((uint32_t*)info.ptr,
-                                      ((uint32_t*)info.ptr) + info.size);
-        return std::unique_ptr<kp::OpAlgoDispatch>{ new kp::OpAlgoDispatch(
-          algorithm, dataVec) };
-    } else if (push_consts.dtype().is(py::dtype::of<std::int32_t>())) {
-        std::vector<int32_t> dataVec((int32_t*)info.ptr,
-                                     ((int32_t*)info.ptr) + info.size);
-        return std::unique_ptr<kp::OpAlgoDispatch>{ new kp::OpAlgoDispatch(
-          algorithm, dataVec) };
-    } else if (push_consts.dtype().is(py::dtype::of<std::double_t>())) {
-        std::vector<double> dataVec((double*)info.ptr,
-                                    ((double*)info.ptr) + info.size);
-        return std::unique_ptr<kp::OpAlgoDispatch>{ new kp::OpAlgoDispatch(
-          algorithm, dataVec) };
-    } else {
-        throw std::runtime_error("Kompute Python no valid dtype supported");
-    }
+    std::vector<byte> dataVec((byte*)info.ptr,
+                                 ((byte*)info.ptr) + info.size);
+    return std::unique_ptr<kp::OpAlgoDispatch>{ new kp::OpAlgoDispatch(
+      algorithm, dataVec) };
 }
 
+class PyTypeContainer : public ABCTypeContainer
+{
+  public:
+    PyTypeContainer(py::object clazz)
+      : clazz_(clazz)
```

Comment:
I don't follow what's going on here, what's class_(...) in this case?

Reply:
class_ should be renamed to dtype. It stores the dtype so we can cast the data back to the correct format in the method data() (line 139). Like wise we return the dtype in data_type() (line 161)

Reply:
renamed for better understanding